src/test/less/parser.test.ts

```diff
@@ -123,7 +144,13 @@ suite('LESS - Parser', () => {
 		let parser = new LESSParser();
 		assertNode('%()', parser, parser._parseFunction.bind(parser));
 		assertNoNode('% ()', parser, parser._parseFunction.bind(parser));
+		
+		assertNode('func(a, b; bar)', parser, parser._parseRuleSetDeclaration.bind(parser));
+		assertNode('func({a: b();}, bar)', parser, parser._parseRuleSetDeclaration.bind(parser));
+		assertNode('func(.(@val) {})', parser, parser._parseRuleSetDeclaration.bind(parser));
```

Comment:
Is this allowed?

This in lessc 3.9 gives me errors:

```less
func(@x) {
  @x();
}

.foo {
  func({a: b();});
}
```

```
 less-test > lessc test.less
ParseError: Unrecognised input in /Users/octref/Code/work/less-test/test.less on line 12, column 10:
11 
12 func(@x) {
13   @x();
```

I thought functions only start with dot, like `.func() {}`?

Reply:
You're thinking of mixins. Mixins start with `.` or `#`, and those are defined in your stylesheet. Functions are defined in JavaScript. They accept a node (or nodes), and return a node. `lighten()` and `darken()` are examples of Less functions. After Less made it easier for users to define functions using `@plugin`, the restriction on functions being only in a property's value was dropped. This way, authors can call a function that returns a ruleset, for example. Less also later added functions that are meant to be called around rules, such as `each()`.